integrations/openai/src/ucai_openai/toolkit.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, List, Optional
+
+from openai.lib._tools import pydantic_function_tool
+from openai.types.chat import ChatCompletionToolParam
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from unitycatalog.ai.client import BaseFunctionClient
+from unitycatalog.ai.utils.client_utils import validate_or_set_default_client
+from unitycatalog.ai.utils.function_processing_utils import (
+    generate_function_input_params_schema,
+    get_tool_name,
+    process_function_names,
+)
+
+
+class UCFunctionToolkit(BaseModel):
```

Comment:
Does this need to be a pydantic model?

Reply:
It doesn't have to be, it's just easier for type validation. Do you have any concerns?

Reply:
Not a big concern, but breaking changes in pydantic.

Reply:
I think we can pin pydantic<3 for now :)

Reply:
Fine to use pydantic for now. Just wondering what the requirements of a toolkit are.

Reply:
I think a normal python class is fine. But the schema extraction process still relies on pydantic, so we're not introducing pydantic as an extra dependency here.